PostFreeSurfer/PostFreeSurferPipelineLongLauncher.sh

```diff
@@ -0,0 +1,256 @@
+opts_AddMandatory '--hiresmesh' 'HighResMesh' 'number' "usually '164', the standard mesh for T1w-resolution data data"
+opts_AddMandatory '--lowresmesh' 'LowResMeshes' 'number' "usually '32', the standard mesh for fMRI data"
+opts_AddMandatory '--subcortgraylabels' 'SubcorticalGrayLabels' 'file' "location of FreeSurferSubcorticalLabelTableLut.txt"
+opts_AddMandatory '--refmyelinmaps' 'ReferenceMyelinMaps' 'file' "group myelin map to use for bias correction"
+opts_AddOptional '--regname' 'RegName' 'name' "surface registration to use, default 'MSMSulc'" 'MSMSulc'
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+log_Check_Env_Var FSLDIR
+verbose_red_echo "---> Starting ${log_ToolName}"
+verbose_echo " "
+verbose_echo " Using environment setting ..."
+verbose_echo "          HCPPIPEDIR: ${HCPPIPEDIR}"
+verbose_echo " "
+IFS=, read -r -a Timepoints <<< "${Timepoint_list[i]}"
+
+fslsub_queue=NONE
+
+case $parallel_mode in
+	FSLSUB) fslsub_queue=$queue ;;
+	BUILTIN) if (( max_jobs<1 )); then max_jobs=2; fi ;;
+	NONE) echo "parallel mode is NONE" ;;
+	*) log_Err_Abort "Unknown parallel mode. Please specify one of FSLSUB, BUILTIN, NONE." ;;
+esac
+
+start_stage=0
+if [ -n "$StartStage" ]; then	
+	case $StartStage in
+		PREP-TP) start_stage=0 ;;
+		PREP-T) start_stage=1 ;;
+		POSTFS-TP1) start_stage=2 ;;
+		POSTFS-T) start_stage=3 ;;
+		POSTFS-TP2) start_stage=4 ;;
+		*) log_Err_Abort "Unrecognized option for start-stage: $StartStage. Must be one of PREP-TP, PREP-T, POSTFS-TP1, POSTFS-T, POSTFS-TP2"
+	esac
+fi
+
+end_stage=4
+if [ -n "$EndStage" ]; then	
+	case $EndStage in
+		PREP-TP) end_stage=0 ;;
+		PREP-T) end_stage=1 ;;
+		POSTFS-TP1) end_stage=2 ;;
+		POSTFS-T) end_stage=3 ;;
+		POSTFS-TP2) end_stage=4 ;;
+		*) log_Err_Abort "Unrecognized option for end-stage: $EndStage. Must be one of PREP-TP, PREP-T, POSTFS-TP1, POSTFS-T, POSTFS-TP2"
+	esac
+fi
+
+#########################################################################################
+# Organising and cleaning up the folder structure
+#########################################################################################
+# ----------------------------------------------------------------------
+log_Msg "Organising and cleaning up the folder structure"
+# ----------------------------------------------------------------------
+
+LongDIR="${StudyFolder}/${Subject}.long.${LongitudinalTemplate}/T1w"
+for TP in ${Timepoints[@]} ; do
+  log_Msg "Organizing the folder structure for: ${TP}"
+  # create the symlink
+	TargetDIR="${StudyFolder}/${TP}.long.${LongitudinalTemplate}/T1w"
+	mkdir -p "${TargetDIR}"
+	ln -sf "${LongDIR}/${TP}.long.${LongitudinalTemplate}" "${TargetDIR}/${TP}.long.${LongitudinalTemplate}"
+
+  # remove the symlink in the subject's folder
+  rm -rf "${LongDIR}/${TP}"
+done
```

Comment:
Should all this really be redone regardless of what the start and end stage are set to?  Should this code be moved into the `...Prep.sh` pipeline?

Reply:
Agree.